generator/reader.go

```diff
@@ -0,0 +1,97 @@
+package generator
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/opencontrol/oscalkit/types/oscal/profile"
+
+	"github.com/opencontrol/oscalkit/types/oscal"
+	"github.com/opencontrol/oscalkit/types/oscal/catalog"
+)
+
+func readOscal(b []byte) (*oscal.OSCAL, error) {
+	r := bytes.NewReader(b)
+	o, err := oscal.New(r)
+	if err != nil {
+		return nil, err
+	}
+	return o, nil
+}
+
+//ReadCatalog ReadCatalog
+func ReadCatalog(b []byte) (*catalog.Catalog, error) {
+
+	o, err := readOscal(b)
+	if err != nil {
+		return nil, fmt.Errorf("cannot read oscal catalog from file %v,", err)
+	}
+	return o.Catalog, nil
+
+}
+
+//ReadProfile reads profile from byte array
+func ReadProfile(b []byte) (*profile.Profile, error) {
+
+	o, err := readOscal(b)
+	if err != nil {
+		return nil, fmt.Errorf("cannot read oscal profile from file. err: %v,", err)
+	}
+	return o.Profile, nil
+}
+
+func isHTTPResource(url *url.URL) bool {
+	return strings.Contains(url.Scheme, "http")
+}
+
+func getName(url *url.URL) string {
+	fragments := strings.Split(url.Path, "/")
+	return (fragments[len(fragments)-1])
+}
+
+func fetchFromHTTPResource(uri *url.URL) ([]byte, error) {
+	c := http.Client{Timeout: 10 * time.Second}
+	resp, err := c.Get(uri.String())
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return nil, fmt.Errorf("cannot read response body %v", err)
+	}
+	return body, nil
+
+}
+
+//GetCatalogFilePath GetCatalogFilePath
+func GetCatalogFilePath(urlString string) (string, error) {
+	uri, err := url.Parse(urlString)
+	if err != nil {
+		return "", fmt.Errorf("invalid URL pattern %v", err)
+	}
+
+	if !isHTTPResource(uri) {
+		return urlString, nil
```

Comment:
if you're executing `oscalkit generate` from a different directory from where the catalogs and profiles exist, if we just return the path as is, this will fail for any profile imports that aren't absolute paths. We should add some logic here to include an absolute base path for any import refs as such.

You're probably going to need to pass both the path of the profile itself and the paths that it imports and check whether or not those imports are absolute paths or relative. `path.IsAbs()` can help.

Reply:
Done